dropshot/tests/fail/bad_endpoint10.stderr

```diff
@@ -0,0 +1,19 @@
+error[E0271]: type mismatch resolving `<String as TypeEq>::This == HttpError`
+  --> $DIR/bad_endpoint10.rs:16:6
+   |
+16 | ) -> Result<HttpResponseOk<()>, String> {
+   |      ^^^^^^
+   |      |
+   |      expected struct `HttpError`, found struct `String`
+   |      required by this bound in `validate_result_error_type`
+
+error[E0277]: the trait bound `fn(Arc<RequestContext<()>>) -> impl Future {<impl From<bad_error_type> for ApiEndpoint<<Arc<RequestContext<()>> as RequestContextArgument>::Context>>::from::bad_error_type}: dropshot::handler::HttpHandlerFunc<_, _, _>` is not satisfied
```

Comment:
We can't actually validate type equality in proc macro context so we generate code that *will* fail *if* there is a type incompatibility. As such, we need to generate both the check *and* the actual code that will be used if everything goes well. In this case we get the more precise, comprehensible error from the check ("expected struct `HttpError`, found struct `String`") and the more baroque (but, of course, accurate) message from the load-bearing code.

I don't see a way to avoid confusion and hope that users elect to fix the easy error messages first (thereby fixing the hard error messages). I don't know why `'static` is underlined. It seems both scary and beside the point.